core/sdk-core/src/main/java/software/amazon/awssdk/core/async/BlockingOutputStreamAsyncRequestBody.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.async;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.reactivestreams.Subscriber;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.exception.NonRetryableException;
+import software.amazon.awssdk.core.internal.util.NoopSubscription;
+import software.amazon.awssdk.utils.CancellableOutputStream;
+import software.amazon.awssdk.utils.async.OutputStreamPublisher;
+
+/**
+ * An implementation of {@link AsyncRequestBody} that allows performing a blocking write of an output stream to a downstream
+ * service.
+ *
+ * <p>See {@link AsyncRequestBody#forBlockingOutputStream(Long)}.
+ */
+@SdkPublicApi
+public class BlockingOutputStreamAsyncRequestBody implements AsyncRequestBody {
+    private final OutputStreamPublisher delegate = new OutputStreamPublisher();
+    private final CountDownLatch subscribedLatch = new CountDownLatch(0);
```

Comment:
Shouldn't it be `new CountDownLatch(1)`?

Reply:
Oh yeah, guess my testing isn't sufficient. Will fix.

Reply:
Fixed.